app/controllers/api/vms_controller.rb

```diff
@@ -302,5 +318,13 @@ def rename_vm(vm, new_name)
     rescue => err
       action_result(false, err.to_s)
     end
+
+    def set_description_vm(vm, new_description)
+      desc = "#{vm_ident(vm)} setting description to #{new_description}"
+      task_id = vm.set_description_queue(User.current_user.userid, new_description)
+      action_result(true, desc, :task_id => task_id)
+    rescue => err
+      action_result(false, err.to_s)
+    end
```

Comment:
Maybe something like this?

```suggestion
    def set_description_resource(type, id, data = {})
      new_description = data&.dig("new_description")&.strip
      api_resource(type, id, "Describing", :supports => :update) do |vm|
        raise BadRequestError, "Must specify a new_description" if new_name.blank?

        {:task_id => vm.set_description_queue(User.current_user.userid, new_description)}
      end
    end
```

Reply:
oops. put this in the wrong spot. think you get the general idea